### PR TITLE
Documentation issues

### LINKED ISSUE #1585: List of inbuilt metrics in score page of label model
## Is your feature request related to a problem? Please describe.
In the documentation page of label model score ([here](https://snorkel.readthedocs.io/en/master/packages/_autosummary/labeling/snorkel.labeling.model.label_model.LabelModel.html?highlight=score#snorkel.labeling.model.label_model.LabelModel.score)), the list of possible metrics are not listed. I had to go through the source to find the inbuilt methods. This the page where I found them :https://github.com/snorkel-team/snorkel/blob/c4e24fbd4b5ff4b9c64298ead2ca0f66d04819d5/snorkel/analysis/metrics.py#L110-L121

It'll also help if examples are present for the user-defined metrics to know how to build one.

## Describe the solution you'd like
The list can be updated in the function doc to show the inbuilt ones present.

### PR DESCRIPTION
**Pull request by TrigonaMinima**
## Description of proposed changes

Added documentation updates for the following:
- List of possible metrics available while scoring
- Model training parameters available

## Related issue(s)

Fixes #1585 

## Test plan

Not applicable

## Checklist

Need help on these? Just ask!

* [x] I have read the **CONTRIBUTING** document.
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] I have run `tox -e complex` and/or `tox -e spark` if appropriate.
* [x] All new and existing tests passed.

### CODE REVIEW

**Comment by TrigonaMinima** [comment]
It's failing because of errors unrelated to my changes.

File: snorkel/labeling/model/label_model.py [hunk 1]
```diff
@@ -810,7 +814,35 @@ def fit(
         class_balance
             Each class's percentage of the population, by default None
         **kwargs
-            Arguments for changing train config defaults
+            Arguments for changing train config defaults.
+
+            n_epochs
```
Comment (henryre) [inline snorkel/labeling/model/label_model.py:819]: Let's point to a page in the documentation (https://snorkel.readthedocs.io/) instead. In case these options update, we don't want to have to remember to change them in two places.

File: snorkel/labeling/model/label_model.py [hunk 2]
```diff
@@ -825,6 +857,7 @@ def fit(
         >>> label_model.fit(L)
         >>> label_model.fit(L, Y_dev=Y_dev)
         >>> label_model.fit(L, class_balance=[0.7, 0.3])
+        >>> label_model.fit(L, n_epochs=200, lr=0.05, l2=0.4, seed=2020)
```
Comment (henryre) [inline snorkel/labeling/model/label_model.py:860]: Let's add one of these keyword arguments to one of the above calls as a demonstration instead so that this doctest isn't too complex.

**Comment by henryre** [comment]
I'll take a look at the errors in CI as well. Looks like a version change for our linting tools.

File: snorkel/labeling/model/label_model.py (hunk 1 above)
Comment (TrigonaMinima) [reply snorkel/labeling/model/label_model.py:819]: Hi, I couldn't find this config anywhere in the snorkel docs that's why I added it here. Could you point me to the page where it is?

**Comment by TrigonaMinima** [comment]
Regarding linting errors, one error was because someone has created the f strings where it's not required. I can fix them in this pull itself if it's fine.

File: snorkel/labeling/model/label_model.py (hunk 1 above)
Comment (TrigonaMinima) [reply snorkel/labeling/model/label_model.py:819]: @henryre

Comment (henryre) [reply snorkel/labeling/model/label_model.py:819]: Great point, it looks like the config isn't included in documentation. This looks like a good place to add it!

**Comment by henryre** [comment]
@TrigonaMinima it looks like there are other linting issues due to a third party package update (possibly mypy itself). Looking into this today!

**Comment by TrigonaMinima** [comment]
Same linting errors unrelated to my changes.

**Comment by henryre** [comment]
@TrigonaMinima just landed linting fixes, so re-running your checks now!

**Comment by henryre** [comment]
@TrigonaMinima if you're happy with the changes, I'll go ahead and merge this in

**Comment by TrigonaMinima** [comment]
@henryre yes we can go ahead with it.
